### PR TITLE
implement slide button

### PR DESCRIPTION
**Pull request by tomkane-dls**
I followed conventions from action and boolean buttons as much as possible

Default rendering from PVI screens:

Screenshot from 2026-05-22 12-41-02

Screenshot from 2026-05-22 12-41-24
Screenshot from 2026-05-22 12-41-32

### CODE REVIEW

File: src/phoebusTheme.ts [hunk 1]
```diff
@@ -19,6 +19,12 @@ export const phoebusTheme = createTheme({
       onColor: "rgba(0,255,0,1)",
       offColor: "rgba(0,100,0,1)"
     },
+    slidebutton: {
+      main: "rgba(0,0,0,0)",
+      contrastText: "#000000",
+      onColor: "rgba(0,255,0,1)",
+      offColor: "rgba(0,100,0,1)"
```
Comment (abigailalexander) [inline src/phoebusTheme.ts]: Phoebus seems to have the default off colour set to grey, instead of dark green

File: src/ui/widgets/SlideButton/slideButton.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+const widgetName = "slidebutton";
+
+const SlideButtonProps = {
+  pvName: StringPropOpt,
+  onState: IntPropOpt,
+  offState: IntPropOpt,
+  onColor: ColorPropOpt,
+  offColor: ColorPropOpt,
+  onLabel: StringPropOpt,
+  offLabel: StringPropOpt,
```
Comment (abigailalexander) [inline src/ui/widgets/SlideButton/slideButton.tsx]: There aren't separate `offLabel` and `onLabel` props, just a single `label` prop. Currently the labels don't show because they aren't being parsed correctly here.

File: src/ui/widgets/SlideButton/slideButton.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import { Switch as MuiSwitch } from "@mui/material";
+import { getPvValueAndName } from "../utils";
+import { useStyle } from "../../hooks/useStyle";
+
+const widgetName = "slidebutton";
+
+const SlideButtonProps = {
+  pvName: StringPropOpt,
+  onState: IntPropOpt,
+  offState: IntPropOpt,
```
Comment (abigailalexander) [inline src/ui/widgets/SlideButton/slideButton.tsx]: `onState`, `offState`, `labelsFromPv` and `backgroundColor` aren't props that exist on this widget. Also, it would be good to add the `visible` prop too

**Review by abigailalexander — CHANGES_REQUESTED** [review]
Thanks for adding this widget in. Looks good overall, just a few comments on props

File: src/ui/widgets/SlideButton/slideButton.tsx (hunk 3 above)
Comment (tomkane-dls) [reply src/ui/widgets/SlideButton/slideButton.tsx]: Ah okay sorry I was comapring with other buttons in cs-web-lib rather than whats defined in Phoebus, I see - I'll compare with ToggleButton now

File: src/ui/widgets/SlideButton/slideButton.tsx (hunk 2 above)
Comment (tomkane-dls) [reply src/ui/widgets/SlideButton/slideButton.tsx]: Ok yes same as below, thanks

File: src/ui/widgets/SlideButton/slideButton.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      ...props,
+      customColors: { onColor: props?.onColor, offColor: props?.offColor }
+    },
+    widgetName
+  );
+
+  const {
+    pvData,
+    enabled = true,
+    visible = true,
```
Comment (abigailalexander) [inline src/ui/widgets/SlideButton/slideButton.tsx:54]: `label` has a default value of "Label", rather than being an empty string/undefined

**Review by abigailalexander — COMMENTED** [review]
Thanks for fixing the props!
I think left-aligning is the best way as this matches what Phoebus does, and ensures everything remains in the same layout as when the .bob file was edited/created. 
Phoebus:
Screenshot from 2026-05-26 13-20-34
cs-web-lib with centre alignment:
Screenshot from 2026-05-26 13-20-29
cs-web-lib with left-alignment:
Screenshot from 2026-05-26 13-29-59
Also noticed here while testing that it wraps text and hides the overflow, so it seems Phoebus isn't too concerned with the text overrun.

**Review by abigailalexander — APPROVED** [review]
All looks great now, thanks!
